### PR TITLE
Merge `dev` into upstream v0.2.1

### PR DESCRIPTION
**Pull request by Vui-Chee**
This PR prepares this repo as a dependency for the new [CDK](https://github.com/okx/xlayer-cdk). It combines custodial assets logic with v0.2.1 updates in the upstream https://github.com/0xPolygon/zkevm-ethtx-manager/tree/v0.2.1. Once merged, a new tag `v0.2.1` will be assigned and referenced from the new CDK repo.

For context, [xlayer-node](https://github.com/okx/xlayer-node) use to contain **ethtxmanager** code, however, the powers that be decided to split ethtxmanager code into a separate repo as a dependency, which invariably led to this PR.

In an effort to make future updates as seamless as possible, most of the new changes are preserved and custom changes are pushed to separate new files.

### Change list
1. change import paths `"github.com/0xPolygonHermez/zkevm-ethtx-manager"` => `"github.com/0xPolygon/zkevm-ethtx-manager"`
2. export custom constructor (with `from` address) for ethtxmanager in separate file `xlayer_ethtxmanager.go`
3. type `monitoredTx` => `zkmanTypes.MonitoredTx` (types file is moved)
4. copied `hex.go` under **hex** folder (these file is extracted from CDK repo)
5. add `GetZkEVMAddressAndL1ChainID()` method to `EthermanInterface`
6. regenerated mocks for interface (`GetZkEVMAddressAndL1ChainID`)

### CODE REVIEW

File: common/common.go
Comment (Vui-Chee) [inline common/common.go:1]: Added by v0.2.1 `0xPolygon/zkevm-ethtx-manager`.

File: .mockery.yml
Comment (Vui-Chee) [inline .mockery.yml:1]: Added by v0.2.1 `0xPolygon/zkevm-ethtx-manager`.

File: ethtxmanager/interfaces.go
Comment (Vui-Chee) [inline ethtxmanager/interfaces.go:1]: Removed in v0.2.1 `0xPolygon/zkevm-ethtx-manager`.

File: ethtxmanager/memstorage.go
Comment (Vui-Chee) [inline ethtxmanager/memstorage.go:1]: Removed in v0.2.1 0xPolygon/zkevm-ethtx-manager.

File: ethtxmanager/monitoredtx.go
Comment (Vui-Chee) [inline ethtxmanager/monitoredtx.go:1]: Moved to `types/interfaces.go` folder in v0.2.1 0xPolygon/zkevm-ethtx-manager.

File: ethtxmanager/monitoredtx_test.go
Comment (Vui-Chee) [inline ethtxmanager/monitoredtx_test.go:1]: Moved to `types/interfaces.go` folder in v0.2.1 0xPolygon/zkevm-ethtx-manager.

**Review by zjg555543 — APPROVED** [review]

**Pull request merged**